perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -226,6 +227,15 @@ def __init__(self, network_spec):
     super(GceNetwork, self).__init__(network_spec)
     self.project = network_spec.project
     name = FLAGS.gce_network_name or 'pkb-network-%s' % FLAGS.run_uri
+
+    # add support for zone, cidr, and separate networks
+    if network_spec.cidr and network_spec.zone:
+      name = FLAGS.gce_network_name or 'pkb-network-%s-%s-%s' % (network_spec.zone,
+                                                                 self.FormatCidrString(network_spec.cidr),
+                                                                 FLAGS.run_uri)
+      FLAGS.gce_subnet_region = util.GetRegionFromZone(network_spec.zone)
+      FLAGS.gce_subnet_addr = network_spec.cidr
```

Comment:
Do not modify flag values? These are global.

Reply:
Fixed. I assumed the order of precedence in configs to be:
- Default (one VPC with subnets in each region)
- Single (one VPC in single region if `--gce_subnet_region` flag is specified and no `cidr` in vm_groups) 
- Multi (one VPC for each vm_group:cidr and one VPC for any/all vm_groups that don't specify cidr explicitly following precedence above )

Flags like `--zone` still apply at the end but now if there's a vm_group:cidr in the config it will create multiple networks in that zone.